### ISSUE TITLE
Let the user pass already opened source file handles to the `materialize` function

### ISSUE THREAD

**Issue by fcharras** [morph-kgc/morph-kgc#286]
#### Is your Feature Request Related to a Problem?
Using rml while stream-processing of big data sources.

#### Describe the Solution you'd Like
Let's assume I have a big file that I don't want to load in memory. Still, I could stream it, cut some small parts, write it to a binary io.BytesIO buffer as a (csv/json/xml) data file, apply rml mapping on the small part, and get the output in another in-memory buffer.

For this to work, I would like to be able to ask the materializer to get the data source, not from a path, but from an arbitrary io object that already exists in my python session, and same for the output.

#### Describe Alternatives you've Considered
Manually parsing my csv/xml/json data structures with idiomatic python tools, not using rml.

**Comment by arenas-guerrero-julian** [comment]
Hi @fcharras ,

In a previous version of the system CSV data was processed in chunks (but was later removed). Users still had to provide a file path, not an io object as you suggest. Would chunk processing providing a file path work for you?

**Comment by fcharras** [comment]
I don't think so, because I suspect my usecases will be too complicated to rely on automatic chunking from a solver, especially so that my data will mostly be json/xml. 

I will not use rml mapping directly on the original files. I will extract sub structures as valid json or xml and run rml mappers on the sub structures only. 

I think this approach is good, because it could let me scale to arbitrary file sizes, while also still letting me use the nice yarrml feature for most of the complicated mapping logic in the sub structures.

**Comment by fcharras** [comment]
In the meantime, it's possible to work around current `materialize` api by temporarily monkey patching the builtin `open` function, which is what I will try now. Although this might work, it is a hack I would be happier if avoided.
